src/libraries/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -528,6 +528,55 @@ public SequencePosition GetPosition(long offset)
             return Seek(offset);
         }
 
+        /// <summary>
+        /// Translate opaque <see cref="SequencePosition"/> value to numerical position offset
+        /// </summary>
+        /// <param name="position"></param>
+        /// <returns></returns>
+        public long GetOffset(SequencePosition position)
+        {
+            bool positionIsNotNull = position.GetObject() != null;
+            BoundsCheck(position, positionIsNotNull);
```

Comment:
I did also a version with check inside method, but I've opted for clear code/code reuse, I don't think this is a "hot path" api.